hardware/detect.py

```diff
@@ -958,65 +984,18 @@ def _main(options):
     parse_dmesg(hrdw, output)
     bios_hp.dump_hp_bios(hrdw)
 
-    if "benchmark_cpu" in options:
-        bm_cpu.cpu_perf(hrdw)
-
-    if "benchmark_mem" in options:
-        bm_mem.mem_perf(hrdw)
-
-    if "benchmark_disk" in options:
-        bm_disk.disk_perf(hrdw,
-                          destructive=options['benchmark_disk_destructive'])
+    if parser.benchmark:
+        if 'cpu' in parser.benchmark:
+            bm_cpu.cpu_perf(hrdw)
+        if 'mem' in parser.benchmark:
+            bm_mem.mem_perf(hrdw)
+        if 'disk' in parser.benchmark:
+            bm_disk.disk_perf(hrdw,
+                              destructive=parser.benchmark_disk_destructive)
```

Comment:
It's a shame this function as a different API.

otherwise we could do 
```
benchmarkers = dict(cpu=bm_cpu.cpu_perf, mem=bm_mem.mem_perf, disk=bm_disk.disk_perf)
...
choices = list(benchmarkers)
...
if parser.benchmark:
   benchmarkers[parser.benchmark](hrdw)
```

Or something like that

Reply:
this approach seems interesting and better than a bunch of ifs :)
I will see what it  can come out from it